src/detectors/builtin/suspiciousMessageMode.ts

```diff
@@ -0,0 +1,117 @@
+import { CompilationUnit } from "../../internals/ir";
+import { forEachExpression } from "../../internals/tact";
+import { MistiTactWarning, Severity } from "../../internals/warnings";
+import { ASTDetector } from "../detector";
+import {
+  AstExpression,
+  AstStructInstance,
+  idText,
+  AstOpBinary,
+} from "@tact-lang/compiler/dist/grammar/ast";
+
+/**
+ * Detects suspicious usage of the `mode` field in `SendParameters` struct instances.
+ *
+ * ## Why is it important?
+ * Incorrect usage of the `mode` field in `SendParameters` can lead to unintended behavior when sending messages,
+ * such as incorrect flags being set, which can cause security vulnerabilities or unexpected contract behavior.
+ *
+ * ## What it checks
+ * - Ensures that the `mode` expression only uses the bitwise OR operator `|`.
+ * - Warns if integer literals are used instead of symbolic constants.
+ * - Warns if the same flag is used multiple times in the mode expression.
+ *
+ * ## Example
+ *
+ * ```tact
+ * // Suspicious usage:
+ * sendParameters.mode = 0x01 | 0x01; // Duplicate flag
+ *
+ * // Correct usage:
+ * sendParameters.mode = FLAG_ONE | FLAG_TWO;
```

Comment:
I would replace this with the complete `send(...)` call using the flags provided in the documentation: https://docs.tact-lang.org/book/message-mode/